sdk/state/payment.go

```diff
@@ -188,9 +191,27 @@ func (c *Channel) validatePayment(ca CloseAgreement) (err error) {
 	if !ca.Details.ConfirmingSigner.Equal(c.localSigner.FromAddress()) && !ca.Details.ConfirmingSigner.Equal(c.remoteSigner) {
 		return fmt.Errorf("close agreement confirmer does not match a local or remote signer, got: %s", ca.Details.ConfirmingSigner.Address())
 	}
+	if !ca.Details.ProposingSigner.Equal(c.localSigner.FromAddress()) && !ca.Details.ProposingSigner.Equal(c.remoteSigner) {
+		return fmt.Errorf("close agreement proposer does not match a local or remote signer, got: %s", ca.Details.ConfirmingSigner.Address())
+	}
+
+	// If the close agreement payment amount is incorrect, error.
+	pa := ca.Details.PaymentAmount
+	if !c.initiatorProposedPayment(ca) {
```

Comment:
💡 We have the `responderSigner` function you can compare the proposer with, which I think is simpler than the `initiatorProposedPayment` function included in this PR.
```suggestion
	proposerIsResponder := ca.Details.ProposingSigner.Equal(c.responderSigner())
	if proposerIsResponder {
```

Reply:
ooh didn't see that cool
https://github.com/stellar/experimental-payment-channels/pull/291/commits/a67f38e91ae613abcd3e289323d490296d70b3c2